_search-plugins/search-pipelines/ml-inference-search-response.md

```diff
@@ -0,0 +1,391 @@
+
+```json
+POST /_plugins/_ml/models/cleMb4kBJ1eYAeTMFFg4/_predict
+{
+  "parameters": {
+    "input": [
+      {
+        ...
+      }
+    ]
+  }
+}
+```
+
+When specifying the `input_map` for an externally hosted model, you can directly reference the `input` field instead of providing its dot path `parameters.input`:
+
+```json
+"input_map": [
+  {
+    "input": "passage_text"
+  }
+]
+```
+
+**Step 2: Run the pipeline**
+
+Run the following query providing the pipeline name in the request:
+
+```json
+GET /my_index/_search?search_pipeline=ml_inference_pipeline_local
+{
+  "query": {
+    "match_all": {
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response confirms that the processor has generated text embeddings in the `passage_embedding` field. The document within `_source` now contains both the `passage_text` and `passage_embedding` fields:
+
+```json
+
+{
+  "took": 288,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.00009405752,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 0.00009405752,
+        "_source": {
+          "passage_text": "hello world",
+          "passage_embedding": [
+            0.017304314,
+            -0.021530833,
+            0.050184276,
+            0.08962978,
+            ...]
+        }
+      }
+      }
+    ]
+  }
+}
+```
+
+### Example: Local model
+
+The following example configures an `ml_inference` search response processor with a local model.
+
+**Step 1: Create a pipeline**
+
+The following example creates an search pipeline for the `huggingface/sentence-transformers/all-distilroberta-v1` local model. The model is a sentence transformer [pretrained model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers) hosted in your OpenSearch cluster.
+
+If you invoke the model using the Predict API, then the request looks like this:
+
+```json
+POST /_plugins/_ml/_predict/text_embedding/cleMb4kBJ1eYAeTMFFg4
+{
+  "text_docs":[ "today is sunny"],
+  "return_number": true,
+  "target_response": ["sentence_embedding"]
+}
+```
+
+Using this schema, specify the `model_input` as follows:
+
+```json
+ "model_input": "{ \"text_docs\": ${input_map.text_docs}, \"return_number\": ${model_config.return_number}, \"target_response\": ${model_config.target_response} }"
+```
+
+In the `input_map`, map the `passage_text` document field to the `text_docs` field expected by the model:
+
+```json
+"input_map": [
+  {
+    "text_docs": "passage_text"
+  }
+]
+```
+
+Because you specified the field to convert into embeddings as a JSON path, you need to set the `full_response_path` to `true` so that the full JSON document is parsed to obtain the input field:
```

Comment:
```suggestion
Because you specified the field to be converted into embeddings as a JSON path, you need to set the `full_response_path` to `true`. Then the full JSON document is parsed in order to obtain the input field:
```